### ISSUE TITLE
Blazor web.config for Gzip/Brotli compression

### ISSUE THREAD

**Issue by guardrex** [dotnet/AspNetCore.Docs#20405]
From @robertmclaws ...

> Hey folks! Wanted to let you know that this sample file presently breaks downloading appsettings.json files for Blazor apps. It seems to be a variant of the dotnet/aspnetcore#26897 issue mentioned earlier. I had to revert my web.config to get my app to work. Any ideas?

https://github.com/dotnet/AspNetCore.Docs/issues/20205#issuecomment-721407229

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: be59faf1-5a20-b7e9-19ff-06589023bb15
* Version Independent ID: 250e7458-3c1d-0117-261d-c03ac136f16a
* Content: [Host and deploy ASP.NET Core Blazor WebAssembly](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/host-and-deploy/webassembly.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/host-and-deploy/webassembly.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
@robertmclaws **UPDATE** ... **_Yes!_** ... I agree. This is a 🐞. My test app failed to load the file. I'll fix this asap on Wednesday. Thanks for the report!

**Comment by guardrex** [comment]
@robertmclaws ...

Oh .... I just looked, and this test app didn't have an `appsettings.json` file. That's why I didn't see it. I added one ... and checked. I'm getting the file here with the updated `web.config` file ...

<img width="469" alt="Capture" src="https://user-images.githubusercontent.com/1622880/98052438-dc014680-1dfb-11eb-82f0-8fda80cd372b.PNG">

Can you try again and confirm your `appsettings.json` file is making it to the `bin\Release\net5.0\publish\wwwroot` folder when you publish the app.

**Comment by cornem** [comment]
Other (default) *.json files also load fine, such as `blazor.boot.json` or `manifest.json`. That said, I didn't test it with `appsettings.json` either...

**Comment by cornem** [comment]
I just tested with an app I have in production that uses this `web.config` file. The compressed `appsettings.json` file loads fine (and is Brotli-compressed) and gets picked up by the `ConfigurationBuilder`.

As I mentioned in the other thread, I'm wondering if the developer might not also be using this `web.config` file in an ASP-NET-hosted Blazor app (won't work), or if the `appsettings.json` file might be in the wrong directory (must be in `wwwroot` as opposed to project root).

**Comment by guardrex** [comment]
If that turns out to be the case @cornem, I think remarks in that section makes sense along the lines of ... 'This `web.config` file is **for standalone WASM apps**. Hosted Blazor apps use a default ASP.NET Core app `web.config` file.'

@robertmclaws ... Let us know if that's what's gone sideways with the guidance.

**Comment by robertmclaws** [comment]
OK, so I just confirmed that it 404's on a standalone Blazor WASM app deployed to Azure. Here's what I did, and all of these edits were made in a live environment, remotely editing the web.config from Visual Studio (so I could verify the results instantly):

1) I started out with the normal version of the web.config that gets autogenerated and deployed with the project: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <staticContent>
      <remove fileExtension=".dat" />
      <remove fileExtension=".dll" />
      <remove fileExtension=".json" />
      <remove fileExtension=".wasm" />
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
      <mimeMap fileExtension=".dll" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json" mimeType="application/json" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
      <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
      <mimeMap fileExtension=".woff2" mimeType="application/font-woff" />
    </staticContent>
    <httpProtocol>
      <customHeaders>
        <add name="Blazor-Environment" value="PROD" />
      </customHeaders>
    </httpProtocol>
    <httpCompression>
      <dynamicTypes>
        <add mimeType="application/octet-stream" enabled="true" />
        <add mimeType="application/wasm" enabled="true" />
      </dynamicTypes>
    </httpCompression>
    <rewrite>
      <rules>
        <rule name="Serve subdir">
          <match url=".*" />
          <action type="Rewrite" url="wwwroot\{R:0}" />
        </rule>
        <rule name="SPA fallback routing" stopProcessing="true">
          <match url=".*" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
          </conditions>
          <action type="Rewrite" url="wwwroot\" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```
That version delivers both my appsettings.json and appsettings.PROD.json files just fine. I want to point out at this point that my app is behaving as expected.

2. I then copied the example version in, modifying it only to add the environment heading to make it grab the right files.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <httpProtocol>
      <customHeaders>
        <add name="Blazor-Environment" value="PROD" />
      </customHeaders>
    </httpProtocol>
    <staticContent>
      <remove fileExtension=".dll" />
      <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
      <mimeMap fileExtension=".blat" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".js.gz" mimeType="application/javascript" />
      <mimeMap fileExtension=".dat.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json.gz" mimeType="application/json" />
      <mimeMap fileExtension=".wasm.gz" mimeType="application/wasm" />
      <mimeMap fileExtension=".blat.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".html.gz" mimeType="text/html" />
      <mimeMap fileExtension=".css.gz" mimeType="text/css" />
      <mimeMap fileExtension=".ico.gz" mimeType="image/x-icon" />
      <mimeMap fileExtension=".svg.gz" mimeType="image/svg+xml" />
      <mimeMap fileExtension=".js.br" mimeType="application/javascript" />
      <mimeMap fileExtension=".dat.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json.br" mimeType="application/json" />
      <mimeMap fileExtension=".wasm.br" mimeType="application/wasm" />
      <mimeMap fileExtension=".blat.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".html.br" mimeType="text/html" />
      <mimeMap fileExtension=".css.br" mimeType="text/css" />
      <mimeMap fileExtension=".ico.br" mimeType="image/x-icon" />
      <mimeMap fileExtension=".svg.br" mimeType="image/svg+xml" />
    </staticContent>
    <httpCompression>
      <dynamicTypes>
        <remove mimeType="text/*" />
        <remove mimeType="application/javascript" />
        <remove mimeType="image/svg+xml" />
      </dynamicTypes>
      <staticTypes>
        <remove mimeType="text/*" />
        <remove mimeType="application/javascript" />
        <remove mimeType="image/svg+xml" />
      </staticTypes>
    </httpCompression>
    <rewrite>
      <outboundRules rewriteBeforeCache="true">
        <rule name="Add Vary Accept-Encoding" preCondition="PreCompressedFile" enabled="true">
          <match serverVariable="RESPONSE_Vary" pattern=".*" />
          <action type="Rewrite" value="Accept-Encoding" />
        </rule>
        <rule name="Add Encoding Brotli" preCondition="PreCompressedBrotli" enabled="true" stopProcessing="true">
          <match serverVariable="RESPONSE_Content_Encoding" pattern=".*" />
          <action type="Rewrite" value="br" />
        </rule>
        <rule name="Add Encoding Gzip" preCondition="PreCompressedGzip" enabled="true" stopProcessing="true">
          <match serverVariable="RESPONSE_Content_Encoding" pattern=".*" />
          <action type="Rewrite" value="gzip" />
        </rule>
        <preConditions>
          <preCondition name="PreCompressedFile">
            <add input="{HTTP_URL}" pattern="\.(gz|br)$" />
          </preCondition>
          <preCondition name="PreCompressedBrotli">
            <add input="{HTTP_URL}" pattern="\.br$" />
          </preCondition>
          <preCondition name="PreCompressedGzip">
            <add input="{HTTP_URL}" pattern="\.gz$" />
          </preCondition>
        </preConditions>
      </outboundRules>
      <rules>
        <rule name="Serve subdir">
          <match url=".*" />
          <action type="Rewrite" url="wwwroot\{R:0}" />
        </rule>
        <rule name="Rewrite brotli file" stopProcessing="true">
          <match url="(.*)"/>
          <conditions>
            <add input="{HTTP_ACCEPT_ENCODING}" pattern="br" />
            <add input="{REQUEST_FILENAME}" pattern="\.(js|dat|dll|json|wasm|blat|htm|html|css|ico|svg)$" />
            <add input="{REQUEST_FILENAME}.br" matchType="IsFile" />
          </conditions>
          <action type="Rewrite" url="{R:1}.br" />
        </rule>
        <rule name="Rewrite gzip file" stopProcessing="true">
          <match url="(.*)"/>
          <conditions>
            <add input="{HTTP_ACCEPT_ENCODING}" pattern="gzip" />
            <add input="{REQUEST_FILENAME}" pattern="\.(js|dat|dll|json|wasm|blat|htm|html|css|ico|svg)$" />
            <add input="{REQUEST_FILENAME}.gz" matchType="IsFile" />
          </conditions>
          <action type="Rewrite" url="{R:1}.gz" />
        </rule>
        <rule name="SPA fallback routing" stopProcessing="true">
          <match url=".*" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
          </conditions>
          <action type="Rewrite" url="wwwroot\" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```

I hit refresh on the appsettings.json and appsettings.PROD.json files (didn;t move anything, just changed the web.config), and they immediately 404'd.

![image](https://user-images.githubusercontent.com/1657085/98170399-e8e56f00-1ebb-11eb-9a85-5a183a50a7dd.png)

Here are my Azure instance settings if that helps: 
![image](https://user-images.githubusercontent.com/1657085/98170251-ad4aa500-1ebb-11eb-97fb-c7e6ba7ef05b.png)

Happy to help track this down, LMK if you need anything else.

**Comment by guardrex** [comment]
Thanks @robertmclaws ... I'll either get back to this tomorrow or Monday ... I hope 🤞. Until then, I'll mark the PR associated with this issue as a DRAFT in case we need updates. I'll get back to you asap ... within the next few days.

**Comment by guardrex** [comment]
@robertmclaws As part of the troubleshooting, you probably should use the Kudu console, navigate to the folder that holds the app's files, and confirm that the compressed versions of the app settings files are there.

**Comment by guardrex** [comment]
**Note to self:** I need to investigate this, too .....

> The stylesheet https://localhost:8011/_framework/scoped.styles.css was not loaded because its MIME type, “text/html”, is not “text/css”.

**Comment by robertmclaws** [comment]
Your explanation of your approach revealed the problem. The web.config posted assumes that it's ONLY a compressed version or ONLY the uncompressed version. I don't think you tested the compressed web.config on an uncompressed app. It's a possibility that someone could switch compression on and off in the build while they are troubleshooting a bug.

You need to replace this line:
```xml
      <remove fileExtension=".dll" />
```

with this:
```xml
      <remove fileExtension=".dll" />
      <remove fileExtension=".json" />
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
      <mimeMap fileExtension=".json" mimeType="application/json" />
      <mimeMap fileExtension=".woff" mimeType="application/font-woff" />
      <mimeMap fileExtension=".woff2" mimeType="application/font-woff" />
```

That will solve the problem and allow the apps to work no matter the compression state. It's tested working in my app.

**Comment by robertmclaws** [comment]
Apologies if my initial version of my past post sounded accusatory, Was not intended.

**Comment by cornem** [comment]
@robertmclaws That's interesting. I suppose it means that those MIME types are not configured by default on the Azure App Service.

I'm using a VM with IIS 10 and then it's no problem to leave out those mappings. When I remove the `br` and `gz` files for `appsettings.json`, it still works because the MIME types for `json`, `woff`, `woff2` files are configured at server level by default.

@guardrex Maybe a good idea to re-add those MIME mappings for apps running on older IIS versions or Azure App Service.

**Comment by guardrex** [comment]
> configured at server level by default

That's what I thought and was going to look into. There's no harm in having them. IIRC, Ben's original file did have them. Perhaps, he was in a similar scenario with Azure. I'll add them back on the PR.

**Comment by robertmclaws** [comment]
Oops, looks like we're not done yet. The "Serve subdir" & "SPA Fallback" rewrite rules break debugging if you put this web.config in the project.

**Comment by guardrex** [comment]
This scenario (rewriting for IIS/Express to serve compressed assets) might be thought of as only supported for non-Development environments. Therefore with debugging being a Development environment-only scenario, the two together might not be a supported scenario (or merely recommended anyway). @captainsafia, thoughts on this?

**Comment by robertmclaws** [comment]
In order for automated deployments to work, you HAVE to have the web.config file in the project. If you want it to work both locally AND deployed, you **MUST** to do config transforms. Unfortunately, they don't work OOB in Blazor projects, you have to set them up manually.

**- Step 1:** Add the following to your Blazor Standalone `.csproj` file:
```xml
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.DotNet.Xdt.Tools" Version="2.0.0" />
  </ItemGroup>

  <Target Name="ApplyXdtConfigTransform" BeforeTargets="_TransformWebConfig">
    <PropertyGroup>
      <_SourceWebConfig>$(MSBuildThisFileDirectory)Web.config</_SourceWebConfig>
      <_XdtTransform>$(MSBuildThisFileDirectory)Web.$(Configuration).config</_XdtTransform>
      <_TargetWebConfig>$(PublishDir)Web.config</_TargetWebConfig>
    </PropertyGroup>
    <Exec Command="dotnet transform-xdt --xml &quot;$(_SourceWebConfig)&quot; --transform &quot;$(_XdtTransform)&quot; --output &quot;$(_TargetWebConfig)&quot;" Condition="Exists('$(_XdtTransform)')" />
  </Target>
```

**- Step 2:** Change your `web.config` to look like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <httpProtocol>
      <customHeaders>
        <add name="Blazor-Environment" value="Development" />
      </customHeaders>
    </httpProtocol>
    <staticContent>
      <remove fileExtension=".dat" />
      <remove fileExtension=".dll" />
      <remove fileExtension=".json" />
      <remove fileExtension=".wasm" />
      <remove fileExtension=".woff" />
      <remove fileExtension=".woff2" />
      <mimeMap fileExtension=".dat" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json" mimeType="application/json" />
      <mimeMap fileExtension=".wasm" mimeType="application/wasm" />
      <mimeMap fileExtension=".woff" mimeType="font/woff" />
      <mimeMap fileExtension=".woff2" mimeType="font/woff2" />     
      <mimeMap fileExtension=".blat" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".js.gz" mimeType="application/javascript" />
      <mimeMap fileExtension=".dat.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json.gz" mimeType="application/json" />
      <mimeMap fileExtension=".wasm.gz" mimeType="application/wasm" />
      <mimeMap fileExtension=".blat.gz" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".html.gz" mimeType="text/html" />
      <mimeMap fileExtension=".css.gz" mimeType="text/css" />
      <mimeMap fileExtension=".ico.gz" mimeType="image/x-icon" />
      <mimeMap fileExtension=".svg.gz" mimeType="image/svg+xml" />
      <mimeMap fileExtension=".js.br" mimeType="application/javascript" />
      <mimeMap fileExtension=".dat.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".dll.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".json.br" mimeType="application/json" />
      <mimeMap fileExtension=".wasm.br" mimeType="application/wasm" />
      <mimeMap fileExtension=".blat.br" mimeType="application/octet-stream" />
      <mimeMap fileExtension=".html.br" mimeType="text/html" />
      <mimeMap fileExtension=".css.br" mimeType="text/css" />
      <mimeMap fileExtension=".ico.br" mimeType="image/x-icon" />
      <mimeMap fileExtension=".svg.br" mimeType="image/svg+xml" />
    </staticContent>
    <httpCompression>
      <dynamicTypes>
        <add mimeType="application/octet-stream" enabled="true" />
        <add mimeType="application/wasm" enabled="true" />
      </dynamicTypes>
    </httpCompression>
    <rewrite>
      <outboundRules rewriteBeforeCache="true">
        <rule name="Add Vary Accept-Encoding" preCondition="PreCompressedFile" enabled="true">
          <match serverVariable="RESPONSE_Vary" pattern=".*" />
          <action type="Rewrite" value="Accept-Encoding" />
        </rule>
        <rule name="Add Encoding Brotli" preCondition="PreCompressedBrotli" enabled="true" stopProcessing="true">
          <match serverVariable="RESPONSE_Content_Encoding" pattern=".*" />
          <action type="Rewrite" value="br" />
        </rule>
        <rule name="Add Encoding Gzip" preCondition="PreCompressedGzip" enabled="true" stopProcessing="true">
          <match serverVariable="RESPONSE_Content_Encoding" pattern=".*" />
          <action type="Rewrite" value="gzip" />
        </rule>
        <preConditions>
          <preCondition name="PreCompressedFile">
            <add input="{HTTP_URL}" pattern="\.(gz|br)$" />
          </preCondition>
          <preCondition name="PreCompressedBrotli">
            <add input="{HTTP_URL}" pattern="\.br$" />
          </preCondition>
          <preCondition name="PreCompressedGzip">
            <add input="{HTTP_URL}" pattern="\.gz$" />
          </preCondition>
        </preConditions>
      </outboundRules>
      <rules>
        <rule name="Rewrite brotli file" stopProcessing="true">
          <match url="(.*)"/>
          <conditions>
            <add input="{HTTP_ACCEPT_ENCODING}" pattern="br" />
            <add input="{REQUEST_FILENAME}" pattern="\.(js|dat|dll|json|wasm|blat|htm|html|css|ico|svg)$" />
            <add input="{REQUEST_FILENAME}.br" matchType="IsFile" />
          </conditions>
          <action type="Rewrite" url="{R:1}.br" />
        </rule>
        <rule name="Rewrite gzip file" stopProcessing="true">
          <match url="(.*)"/>
          <conditions>
            <add input="{HTTP_ACCEPT_ENCODING}" pattern="gzip" />
            <add input="{REQUEST_FILENAME}" pattern="\.(js|dat|dll|json|wasm|blat|htm|html|css|ico|svg)$" />
            <add input="{REQUEST_FILENAME}.gz" matchType="IsFile" />
          </conditions>
          <action type="Rewrite" url="{R:1}.gz" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```
This version combines the static content remappings from the default `web.config` that the build process adds if one is not present, with the Rewrite rules that will not break the app locally.

**- Step 3:** Add a `web.{ENVIRONMENT}.config` and add the following:
```xml
<?xml version="1.0" encoding="utf-8"?>

<!-- For more information on using web.config transformation visit https://go.microsoft.com/fwlink/?LinkId=125889 -->

<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">

  <system.webServer>
    <httpProtocol>
      <customHeaders>
        <add xdt:Transform="SetAttributes" xdt:Locator="Match(name)" name="Blazor-Environment" value="{YOUR ENVIRONMENT HERE}" />
      </customHeaders>
    </httpProtocol>
    <httpCompression xdt:Transform="Replace">
      <dynamicTypes>
        <remove mimeType="text/*" />
        <remove mimeType="application/javascript" />
        <remove mimeType="image/svg+xml" />
      </dynamicTypes>
      <staticTypes>
        <remove mimeType="text/*" />
        <remove mimeType="application/javascript" />
        <remove mimeType="image/svg+xml" />
      </staticTypes>
    </httpCompression>
    <rewrite>
      <rules>
        <rule xdt:Transform="InsertBefore(/configuration/system.webServer/rewrite/rules/rule[@name='Rewrite brotli file'])"
          name="Serve subdir">
          <match url=".*" />
          <action type="Rewrite" url="wwwroot\{R:0}" />
        </rule>
        <rule xdt:Transform="InsertAfter(/configuration/system.webServer/rewrite/rules/rule[@name='Rewrite gzip file'])"
          name="SPA fallback routing" stopProcessing="true">
          <match url=".*" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
          </conditions>
          <action type="Rewrite" url="wwwroot\" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>

</configuration>
```
This version adds the environment headers, replaces the compression section, and inserts nodes at the beginning and end of the rewrite rules, so that the deployed version matches what is currently in the documentation.

That allows the app to work locally for debugging, and have the web.config transformed appropriately for deployment environments... while also working with the appsettings environment transform requirements.

Hope that makes it easy for you folks to capture in the documentation. HTH!

**Comment by guardrex** [comment]
I opened https://github.com/dotnet/AspNetCore.Docs/issues/20445.

Yes, I'm aware of all of that, but my point stands. The two configurations with **_one, untransformed web.config file_** across environments is apparently not a supported scenario OOB for debugging. That's why I asked @captainsafia to comment.

WRT your remark, the production `web.config` can be swapped into the deployment assets prior to deployment, changed on the hosting system after deployment but before starting the app pool, or exist/persist on the hosting system continuously (i.e., not deployed with the app). Transforms are one of several approaches that devs can find success with, including in automated deployment scenarios.

WRT to covering the transform approach, that's an advanced scenario given our current, ordinary level of coverage. I don't know if management would like it covered or not, so I opened that issue and marked it for triage. Given that we're right on the edge of 5.0 GA, I don't think that we'll get a quick answer. Then with the ⛄ *holidays* 🎁 fast approaching, it might be triaged early next year. Hard to say ... everyone is 🏃😅 like crazy right now. We'll keep an :ear: on the issue to see if they give it any ❤️ ... thanks for the suggestion.

**Comment by captainsafia** [comment]
> Compression with an untransformed web.config file across environments is apparently not a supported scenario OOB for debugging. That's why I asked @captainsafia to comment.

Trying to catch up with the thread. Is there a repro for a project with the web.config that fails to debug?

**Comment by robertmclaws** [comment]
👋🏻 @captainsafia! Just use [either of the web.configs specified at the top of this post](https://github.com/dotnet/AspNetCore.Docs/issues/20405#issuecomment-721988548) in a Blazor project and debug locally.

Appreciate all of the quick responses and the engagement, you folks are awesome!

@Guardrex unfortunately none of the approaches you specified in your last post are tenable from an operations standpoint in anything but hobbyist situations. When a `web.config` does not exist in the project, the `dotnet publish` process generates a new one, which overwrites whatever you've previously uploaded. (I just confirmed this in my own Azure DevOps environment.) Any other support mechanism (Azure DevOps tasks, post-deployment scripts, manual intervention) are actually *more* complicated than the config transform process I outlined, which conceptually most developers should be familiar with.

IMO, I think we've uncovered that compression itself is an advanced scenario that requires its own in-depth article. Given all of the other constraints on the next few weeks, and that most developers would likely rather be able to debug their project than have optimized compression... it might make sense to pull compression from the docs for now. As you mentioned in the other issue, I'm happy to write a blog post about it to link from Awesome Blazor.

**Comment by guardrex** [comment]
Publish with ...

```xml
<PublishIISAssets>false</PublishIISAssets>
```

... and hopefully it won't produce one or leave the one in the `publish` folder alone. 🤞

IIRC, set to `true` to take a custom file from the project root (maybe 🤞 ... it's a been a while since I played with that property).

*hobbyist* :smile:

@captainsafia will let us know if the situation is so dire that we'd need to pull coverage ... or NOTE ... or WARNING ... we have options to leave the current coverage and surface something to readers on this.

WRT publishing something on the transform scenario, I think a blog post+Awesome Blazor would be best for *FAST* coverage. The issue that I opened might not get triaged for a while. Even if it becomes a work item, I have a feeling that it will get a low priority. If so, that would definitely push it to next year.
